asset-transfer-private-data/chaincode-go/privateAssetTransfer.go

```diff
@@ -0,0 +1,469 @@
+type AssetPrivateDetails struct {
+	ID             string `json:"asset_id"`
+	AppraisedValue int    `json:"appraisedValue"`
+}
+
+const assetCollection = "assetCollection"
+
+type SmartContract struct {
+	contractapi.Contract
+}
+
+// CreateAsset creates a new asset by placing the main asset details in the assetCollection
+// that can be read by both organizations. The appraisal value is stored in the owners org specific collection.
+func (s *SmartContract) CreateAsset(ctx contractapi.TransactionContextInterface) error {
+
+	// Get new asset from transient map
+	transMap, err := ctx.GetStub().GetTransient()
+	if err != nil {
+		return fmt.Errorf("Error getting transient: " + err.Error())
+	}
+
+	// Asset properties are private, therefore they get passed in transient field
+	transientAssetJSON, ok := transMap["asset_properties"]
+	if !ok {
+		return fmt.Errorf("asset not found in the transient map")
+	}
+
+	type assetTransientInput struct {
+		ObjectType 		 	string `json:"object_type"`  //docType is used to distinguish the various types of objects in state database
+		ID             	string `json:"asset_id"`
+		Color          	string `json:"color"`
+		Size           	int    `json:"size"`
+		AppraisedValue 	int    `json:"appraisedValue"`
+	}
+
+	var assetInput assetTransientInput
+	err = json.Unmarshal(transientAssetJSON, &assetInput)
+	if err != nil {
+		return fmt.Errorf("failed to unmarshal JSON: %s", err.Error())
+	}
+
+	if len(assetInput.ObjectType) == 0 {
+		return fmt.Errorf("object_type field must be a non-empty string")
+	}
+	if len(assetInput.ID) == 0 {
+		return fmt.Errorf("asset_id field must be a non-empty string")
+	}
+	if len(assetInput.Color) == 0 {
+		return fmt.Errorf("color field must be a non-empty string")
+	}
+	if assetInput.Size <= 0 {
+		return fmt.Errorf("size field must be a positive integer")
+	}
+	if assetInput.AppraisedValue <= 0 {
+		return fmt.Errorf("AppraisedValue field must be a positive integer")
+	}
+
+	// Check if asset already exists
+	assetAsBytes, err := ctx.GetStub().GetPrivateData(assetCollection, assetInput.ID)
+	if err != nil {
+		return fmt.Errorf("Failed to get asset: " + err.Error())
+	} else if assetAsBytes != nil {
+		fmt.Println("This asset already exists: " + assetInput.ID)
+		return fmt.Errorf("This asset already exists: " + assetInput.ID)
+	}
+
+	// Get ID of submitting client identity
+	clientID, err := ctx.GetClientIdentity().GetID()
+	if err != nil {
+		return fmt.Errorf("failed to get verified OrgID: %s", err.Error())
+	}
+
+	// Make submitting client the owner
+	asset := &Asset{
+		ObjectType: assetInput.ObjectType,
+		ID:       	assetInput.ID,
+		Color:      assetInput.Color,
+		Size:       assetInput.Size,
+		Owner:      clientID,
+	}
+	assetJSONasBytes, err := json.Marshal(asset)
+	if err != nil {
+		return fmt.Errorf(err.Error())
+	}
+
+	// Save asset to private data collection
+	err = ctx.GetStub().PutPrivateData(assetCollection, assetInput.ID, assetJSONasBytes)
```

Comment:
I see  marble sample had two PutPrivateData calls, but I am thinking if we can simplify

1. put public details of asset using  `ctx.GetStub().PutState()` with ID as key
1. put private data into orgCollection with `PutPrivateData()`

Reply:
For simplicity sake:  can we remove the first collection from collections_config.json: `"name": "assetCollection"`.

Thoughts?